bin/format.js

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+
+const articles = findArticles(path.join(__dirname, '..'));
+const changes = {};
+const rules = [
+    null,
+    rule_1_boundaryWhitesapce,
+    rule_2_atxHeaders,
+    rule_3_noTabs,
+    rule_4_smartCharacters
+];
+
+const ruleArgIndex = process.argv.findIndex(a => a === '-r' || a === '--rule') + 1;
+const rulesToCheck = ruleArgIndex === 0 ? null : process.argv[ruleArgIndex].split(',').map(r => parseInt(r));
+
+articles.forEach(article => format(article));
+
+let changeCountTotal = 0;
+let changedFilesTotal = [];
+
+for (let i = 0; i < rules.length; i++) {
+    if (changes[i] === undefined)
+        continue;
+
+    const changeCount = Object.values(changes[i]).reduce((sum, c) => sum + c, 0);
+    const changedFiles = Object.entries(changes[i]).filter(c => c[1] > 0).map(c => c[0]);
+    const indent = ' '.repeat(4 - i.toString().length);
+
+    changeCountTotal += changeCount;
+    changedFilesTotal = changedFilesTotal.concat(changedFiles);
+
+    console.log(`\x1b[30m\x1b[43mRULE ${i}\x1b[0m${indent}Made \x1b[33m${changeCount.toString().padStart(3)}\x1b[0m changes to \x1b[33m${changedFiles.length.toString().padStart(3)}\x1b[0m articles`);
+};
+
+const changedFilesCount = changedFilesTotal.filter((f, i, files) => files.indexOf(f) === i).length;
+
+console.log(`\x1b[30m\x1b[42mDONE!\x1b[0m    Made \x1b[32m${changeCountTotal.toString().padStart(3)}\x1b[0m changes to \x1b[32m${changedFilesCount.toString().padStart(3)} / ${articles.length}\x1b[0m articles`);
+
+function rule_1_boundaryWhitesapce(content, meta, log) {
```

Comment:
document what this function does, because definition name does not exactly state what the function is supposed to do.